crates/cheatcodes/src/inspector.rs

```diff
@@ -1899,6 +1919,11 @@ where {
 
         info!("running call in zkEVM {:#?}", call);
         let zk_persist_nonce_update = self.zk_persist_nonce_update.check();
+
+        // Note(zk): Clear injected factory deps here even though it's actually used in broadcast.
+        // To be consistent with where we clear factory deps in try_create_in_zk.
+        self.zk_use_factory_deps.clear();
```

Comment:
Why are we clearing them here?

Reply:
We could clear it in call_with_executor where we inject them but to be consistent with try_create_in_zk. We are clearing them because we weren't cleaning them before.

Reply:
Yeah it's fine to clear them here, just was wondering why all of a sudden.